src/filter.js

```diff
@@ -0,0 +1,347 @@
+// Add Place Filter
+
+import axios from "axios";
+
+let buttonValue = {
+    Spring: false,
+    Public: false,
+    Partner: false,
+    Filtered: false,
+    Hot: false,
+    Cold: false,
+    WiFi: false,
+    Accessible: false,
+    Tap: false,
+    Beauty: false,
+    Restaurant: false,
+    Accommodation: false,
+    Shop: false,
+    Yourself: false,
+    Staff: false,
+}
+
+const buttonTags = {
+    Filtered: 1,
+    Hot: 2,
+    Cold: 3,
+    WiFi: 4,
+    Accessible: 5,
+    Tap: 6,
+    Beauty: 7,
+    Restaurant: 8,
+    Accommodation: 9,
+    Shop: 10,
+}
+
+const buttonCategories = {
+    Spring: 1,
+    Public: [0, 2, 3],
+    Partner: 4
+}
+
+
+let setPlaces;
+let setMap;
+let setMaps;
+let clickBool = false;
+let updateBool = false;
+
+export function createFilter(map, maps, setTaps) {
+    setMap = map;
+    setMaps = maps;
+    setPlaces = setTaps;
+
+    let control = document.createElement('div');
+    const filterButtonUI = menuButtonUi();
+    control.appendChild(filterButtonUI);
+    setMap.controls[maps.ControlPosition.RIGHT_BOTTOM].push(control);
+    const filterBoxDiv = document.createElement("div");
+    const filterUI = filterBoxUi(filterBoxDiv);
+    filterParametersUi(filterUI);
+
+    filterBoxDiv.appendChild(filterUI);
+
+    filterButtonUI.addEventListener("click", () => {
+        filteringMenuAction(filterBoxDiv)
+    });
+
+    // Update map when dragged
+    const updateBoxDiv = document.createElement("div");
+    updateBoxDiv.style.cursor = "pointer";
+    const updateUI = updateBoxUI();
+    updateBoxDiv.appendChild(updateUI);
+    setMaps.event.addListener(map, 'dragend', async () => {
+        if (!updateBool) {
+            setMap.controls[setMaps.ControlPosition.TOP_CENTER].push(updateBoxDiv);
+            updateBool = !updateBool;
+        }
+    });
+    setMaps.event.addListener(map, 'zoom_changed', async () => {
+        if (!updateBool) {
+            setMap.controls[setMaps.ControlPosition.TOP_CENTER].push(updateBoxDiv);
+            updateBool = !updateBool;
+        }
+    });
+}
+
+function menuButtonUi() {
+    // Button UI
+    const divButton = document.createElement("div");
+    divButton.id = "filter";
```

Comment:
I want to convert this file into a react component (e.g. https://github.com/mymizu/mymizu-web/blob/main/src/components/Metrics.jsx) , and seperate the styling out to a css file if possible!

Please let me know if I can explain how to approach !